File: LeetCode/2367. Number of Arithmetic Triplets/solution.cpp
/**
 * Runtime: 4 ms
 * Memory Usage: 8.7 MB 
 */
class Solution {
public:
    int arithmeticTriplets(vector<int>& nums, int diff) {
        unordered_set<int> rec;
        int ans = 0;
        for (const auto& num: nums) {
            if (rec.count(num - 2 * diff) && rec.count(num - diff)) {
                ++ ans;
            }
            rec.emplace(num);
        }
        return ans;
    }
};

File: LeetCode/2367. Number of Arithmetic Triplets/solution.py
"""
Runtime: 32 ms
Memory Usage: 15 MB
"""
class Solution:
    def arithmeticTriplets(self, nums: List[int], diff: int) -> int:
        ans, rec = 0, set()
        for num in nums:
            if (num - 2 * diff) in rec and (num - diff) in rec:
                ans += 1
            rec.add(num)
        return ans
       
